openslides_backend/action/generics.py

```diff
@@ -164,33 +156,24 @@ def update_action_prepare_dataset(self, payload: ActionPayload) -> DataSet:
             relation_fields = []
             additional_instance_fields: Dict[str, Set[str]] = defaultdict(set)
             for field_name, field in self.model.get_relation_fields():
-                for instance_field in instance.keys():
-                    if field_name == instance_field:
-                        if field.structured_relation:
-                            if instance.get(field.structured_relation[0]) is not None:
-                                raise ActionException(
-                                    "You must not try to update both a relation field "
-                                    "with structured_relation and its corresponding "
-                                    "foreign key field."
-                                )
-                        relation_fields.append((field_name, field))
-                    elif isinstance(field, BaseTemplateRelationField):
-                        regex = (
-                            r"^"
-                            + field_name[: field.index]
-                            + r"\$"
-                            + r"(\d+)"
-                            + field_name[field.index :]
-                            + r"$"
-                        )
-                        match = re.match(regex, instance_field)
-                        if not match:
-                            continue
+                if field_name in instance:
+                    if field.structured_relation:
+                        if instance.get(field.structured_relation[0]) is not None:
+                            raise ActionException(
+                                "You must not try to update both a relation field "
+                                "with structured_relation and its corresponding "
+                                "foreign key field."
+                            )
+                    relation_fields.append((field_name, field))
+                elif isinstance(field, BaseTemplateField):
+                    structured_fields = self.get_structured_fields_in_instance(
+                        field_name, field, instance
+                    )
+                    for instance_field, replacement in structured_fields:
```

Comment:
Here the same: replacement or substitute (maybe use always the same term) must not contain other chars than digits.